addons/mail/static/src/core/common/core.scss

```diff
@@ -41,3 +41,14 @@
 .o-z-index-1 {
     z-index: 1;
 }
+
+a.o_mail_redirect, a.o_channel_redirect {
+    background: rgba($primary, .2) !important;
+    @extend .btn;
+    @extend .btn-sm;
+    @extend .btn-link;
+    @extend .align-baseline;
+    @extend .fw-normal;
+    @extend .px-1;
+    @extend .py-0;
+}
```

Comment:
@zel-odoo @alexkuhn You should never do that. Arguably it could be ok with pure utility classes like the last four because they should not be used too much in other selectors but definitely not ok with `.btn`.

Here this is breaking the website navbar for example (actually, should this file even impact the frontend?):
![image](https://github.com/odoo/odoo/assets/10338094/1eaae5c7-9e27-4f45-9e66-9b351b1ad56f)
=> The navbar links are now colored, they should not be.

This is because we have such a rule in our SCSS: `.a:not(.btn) {`. Because of your `@extend`, this became: `a:not(.btn):not(.o_mail_redirect):not(.o_channel_redirect)`
- First, you really do not want every app of Odoo complexifying all selectors like that... the resulting CSS will be awful
- Second, this simply breaks the selector: `a:not(.btn):not(.stuff)` has a higher specificity than `a:not(.btn)` which is why our navbar is now broken.
- This has been know to be inefficient. When we switched to sass instead of less, just removing all the `@extend` call speed up CSS generation by like 5x.

Really, `@extend` should only be used with placeholder selectors (`%aaa`) which you can control more easily.

Would you fix this please? :pray: (can't you use the classes in the XML directly?)

Reply:
Hello, we can't add in XML directly because the targeted content is dynamic (it might be present on any message and multiple times: this targets all `@mentions`).

What we'd like is to style them while reusing style defined in these classnames. What can we use instead of `@extend` for that purpose? Thanks!

Reply:
> we can't add in XML directly because the targeted content is dynamic: it might be present on any message and multiple times: this targets all @mentions

I don't understand what you mean: something is adding that `o_mail_redirect` class somehow. That code could add the other classes that goes with it as well, no?

Reply:
This classname is added when posting a new message, the message will contain indefinitely this classname, as a way to flag some internal content as a mention.

We definitely don't want to put all the classnames in the message, otherwise whenever we change this style, we should make migration scripts to adapt all mentions in all previously sent messages in the DB.

Using `o_mail_redirect` is a minimal footprint for the pure logic of mention. For the style, we'd like to rely solely on (S)CSS, so that it's easier to maintain.

Reply:
> Using o_mail_redirect is a minimal footprint for the pure logic of mention.

Well, I understand and that would be nice but I don't think you will be able to do that properly then. For this to be possible, bootstrap should have exposed a mixin or a placeholder selector to extend so that this would not rely on the style of "buttons in different context" (since what you want is "the base style of buttons"). Can't you rely on some JS? "When displaying (not saving) the message, find the `o_mail_redirect` element and add those classes on them". It's less pretty but it is basically the same as "When displaying (not saving) the message, apply the same CSS style as those elements on it"... without the related bugs.

The alternative would be purely copying the style. Also `.btn.btn-sm.btn-link` is maybe kinda the same a just link elements no? Maybe copying the style of a `<a src="align-baseline fw-normal px-1py-0">` would not be so awful.

Also, if we are talking about the style of message mentions... I really think this file should not be in the frontend then (or at least not that rule?)

Reply:
> Can't you rely on some JS? "When displaying (not saving) the message, find the `o_mail_redirect` element and add those classes on them".

Ok will make a PR to fix it that way. A bit sad to add some custom LOCs in JS to cascade the style to a specific element.

Reply:
Follow-up PR: https://github.com/odoo/odoo/pull/138108
In the end not injecting classnames in JS, because it's too challenging with web editor / composer / message content post-processing that is already markup.
So in the end, solution is just relying on `button-variant` mixin